staging/src/k8s.io/client-go/util/workqueue/queue_test.go

```diff
@@ -321,3 +323,40 @@ func TestQueueDrainageUsingShutDownWithDrainWithDirtyItem(t *testing.T) {
 
 	finishedWG.Wait()
 }
+
+type bigObject struct {
+	data []byte
+}
+
+// Created outside TestKubeLeak so it won't be GCed.
+var leakQueue = workqueue.New()
+
+// TestGarbageCollection ensures that objects that are added then removed from the queue are
+// able to be garbage collected.
+func TestGarbageCollection(t *testing.T) {
+	c := &bigObject{}
+	mustGarbageCollect(t, c)
+	leakQueue.Add(c)
+	o, _ := leakQueue.Get()
+	leakQueue.Done(o)
+}
+
+// mustGarbageCollect asserts than an object was garbage collected by the end of the test.
+// The input must be a pointer to an object.
+func mustGarbageCollect(t *testing.T, i interface{}) {
+	t.Helper()
+	collected := false
+	runtime.SetFinalizer(i, func(x interface{}) {
+		collected = true
+	})
+	t.Cleanup(func() {
+		if err := wait.PollImmediate(time.Millisecond*100, time.Second*5, func() (done bool, err error) {
+
+			// Trigger GC explicitly, otherwise we may need to wait a long time for it to run
+			runtime.GC()
+			return collected, nil
```

Comment:
can referencing `collected` here and writing it in the finalizer cause a data race?